### PR TITLE
Fixes issue 24868 Clean code: fix unlikely argument and remove some unused variables. 

### PR DESCRIPTION
**Pull request by escay**
ManagedBeanManagerImpl and JdbcConnPoolAppStatsProvider need specific review.
I did not add a nullcheck for SimpleJndiName in JdbcConnPoolAppStatsProvider, since the constructor already validates it.

```
## Unlikely argument type SimpleJndiName for remove(Object) on a Map<String,NamingObjectProxy>	ManagedBeanManagerImpl.java	/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean	line 338	Java Problem
## Unlikely argument type for equals(): SimpleJndiName seems to be unrelated to String	JdbcConnPoolAppStatsProvider.java	/jdbc-runtime/src/main/java/org/glassfish/jdbc/pool/monitor	line 90	Java Problem
## Unlikely argument type for equals(): SimpleJndiName seems to be unrelated to String	JdbcConnPoolAppStatsProvider.java	/jdbc-runtime/src/main/java/org/glassfish/jdbc/pool/monitor	line 109	Java Problem
## Unlikely argument type for equals(): SimpleJndiName seems to be unrelated to String	JdbcConnPoolAppStatsProvider.java	/jdbc-runtime/src/main/java/org/glassfish/jdbc/pool/monitor	line 124	Java Problem
## Unlikely argument type for equals(): SimpleJndiName seems to be unrelated to String	JdbcConnPoolAppStatsProvider.java	/jdbc-runtime/src/main/java/org/glassfish/jdbc/pool/monitor	line 136	Java Problem
```

Full list of items is listed in issue #24868

### CODE REVIEW

**Comment by escay** [comment]
I will have a look at the 
```
[2024-03-21T11:49:02.051Z] [INFO] There are 2 errors reported by Checkstyle 10.12.4 with org/glassfish/qa/config/checkstyle/checkstyle.xml ruleset.
[2024-03-21T11:49:02.051Z] [ERROR] src/main/java/org/glassfish/appclient/client/acc/HTTPInputArchive.java:[125] (regexp) RegexpSingleline: Line has trailing spaces.
[2024-03-21T11:49:02.051Z] [ERROR] src/main/java/org/glassfish/appclient/client/acc/HTTPInputArchive.java:[130] (regexp) RegexpSingleline: Line has trailing spaces.
```
build failure.

**Comment by escay** [comment]
I probably made something in 
```
[2024-03-21T13:02:35.435Z]      [exec] input file=/home/jenkins/agent/workspace/_test-using-jenkinsfile_PR-24870/appserver/tests/appserv-tests/test_resultsValid.xml
[2024-03-21T13:02:35.435Z]      [exec] 
[2024-03-21T13:02:35.435Z]      [exec] ************************
[2024-03-21T13:02:35.435Z]      [exec] PASSED=   31
[2024-03-21T13:02:35.435Z]      [exec] ------------  =========
[2024-03-21T13:02:35.435Z]      [exec] FAILED=   1
```
fail. I will look into it.

**Comment by escay** [comment]
Build server tests now succeed (did someone start a new build? was the cause the "java.net.BindException: Address already in use"?) And I also ran the ant tests locally and they succeed:
```
     [exec] input file=/mnt/d/gf-escay/glassfish/appserver/tests/appserv-tests/test_resultsValid.xml
     [exec] ************************
     [exec] PASSED=   32
     [exec] ------------  =========
     [exec] FAILED=   0
```

**Comment by pzygielo** [comment]
Commit comment and PR title suggest that this fixes some issue.

GH provides https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue - with specific syntax for that to happen automatically.

File: appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java [hunk 1]
```diff
@@ -335,7 +335,7 @@ public void unloadManagedBeans(Application app) {
                             + " with jndi name " + jndiName, ne);
                     }
                 } else {
-                    appClientManagedBeans.remove(jndiName);
+                    appClientManagedBeans.remove(jndiName.toString());
```
Comment (dmatej) [inline appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java:338]: Wow, good catch!

File: appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/JMSDestinationDefinitionDeployer.java [hunk 2]
```diff
@@ -43,7 +35,13 @@
 import org.jvnet.hk2.config.TransactionFailure;
 import org.jvnet.hk2.config.types.Property;
 
-import static com.sun.appserv.connectors.internal.api.ConnectorsUtil.deriveResourceName;
+import com.sun.appserv.connectors.internal.api.ConnectorConstants;
```
Comment (dmatej) [inline appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/JMSDestinationDefinitionDeployer.java:38]: Imports should be ordered by alphabet. Static imports should be first or last, but together, that is ok.

File: appserver/web/web-core/src/main/java/org/apache/catalina/startup/ExpandWar.java [hunk 3]
```diff
@@ -297,30 +297,14 @@ public static boolean copy(File src, File dest) {
             if (fileSrc.isDirectory()) {
                 result = copy(fileSrc, fileDest);
             } else {
-                FileChannel ic = null;
-                FileChannel oc = null;
-                try {
-                    ic = (new FileInputStream(fileSrc)).getChannel();
-                    oc = (new FileOutputStream(fileDest)).getChannel();
+                try (FileChannel ic = (new FileInputStream(fileSrc)).getChannel();
```
Comment (dmatej) [inline appserver/web/web-core/src/main/java/org/apache/catalina/startup/ExpandWar.java:300]: I'm not sure now, is closing the channel enough to close the input stream? Is it possible that GC would close the input stream before the channel (no reference to the IS)?

**Review by dmatej — APPROVED** [review]
Approved, except that ordering of imports, but as we still don't have checkstyle rule enabled for that I can accept that even as it is.

File: appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/JMSDestinationDefinitionDeployer.java (hunk 2 above)
Comment (escay) [reply appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/JMSDestinationDefinitionDeployer.java:38]: Sorry I assumed the ee4j Eclipse formatter took care of the order. I will minimise import statement changes in the future.

File: appserver/web/web-core/src/main/java/org/apache/catalina/startup/ExpandWar.java (hunk 3 above)
Comment (escay) [reply appserver/web/web-core/src/main/java/org/apache/catalina/startup/ExpandWar.java:300]: I am not sure. I did compare it to the latest real catalina Expandwar code. They used the same approach, so I assumed it was safe. Note that latest ExpandWar in catalina has some other fixes that are not in this fork.
https://github.com/apache/tomcat/blob/main/java/org/apache/catalina/startup/ExpandWar.java

**Review by arjantijms — APPROVED** [review]

**Comment by dmatej** [comment]
Is it possible you introduced a new error somehow - sometimes Jenkins fails this test on JDK21, that did not happen before:
```
org.opentest4j.AssertionFailedError: expected: <5> but was: <3>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:531)
	at com.sun.enterprise.resource.pool.ConnectionPoolTest.assertResourcesSize(ConnectionPoolTest.java:437)
	at com.sun.enterprise.resource.pool.ConnectionPoolTest.basicConnectionPoolMultiThreadedTest(ConnectionPoolTest.java:314)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```
It is quite often but not well reproducible, seems like some race condition. Could be caused also by some other change, but this is the closest one.

**Comment by escay** [comment]
I added the test about a month ago, the test might also be incorrect / my reading / understanding of the code might be incorrect.
I have not tested with JDK21 myself, but I can try it out tomorrow.

If it is needed line ConnectionPoolTest.java:314 could be disabled / removed.

**Comment by escay** [comment]
Installed OpenJDK21U-jdk_x64_windows_hotspot_21.0.3_9.zip in windows.

Ran 20 times "mvn test -offline" in module "connectors-runtime", ConnectionPoolTest never failed.
Removed the Thread sleep 20 nanos -> test fails with: org.opentest4j.AssertionFailedError: expected: <5> but was: <3>
Removed the Thread sleep 20 nanos and increased tasks from 100 to 10.000 -> test succeeds

I think the test with line 314 is 'CPU / machine' dependent (I use a 6 year old i7 8700 cpu).

Two fixes I can think of: 

1. Remove the test line, this would be ok, because the poolResizeQuantityTest and the basicConnectionPoolTest already test the values while controlling when resources are returned to the pool
2. Provide some information to the tasks created in runTheTasks to make the first 5 threads occupy the pool for a bit

I would opt for option 1 to keep it simple, the test is already complicated enough.

**Comment by escay** [comment]
Created issue https://github.com/eclipse-ee4j/glassfish/issues/24931 to improve the test, creating a pull request.

**Pull request merged**